### PR TITLE
SDFSOF-68: RPC Actions - update Kotlin Reference Server to use new endpoint

### PR DESCRIPTION
**Pull request by rkharevych**
<details>
  <summary>PR Checklist</summary>

### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `paymentservice.stellar`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
</details>

### What

Updated Kotlin Reference Server to use new RPC endpoint

### Why

All transaction updates should be done using new RPC actions endpoint. In scope of this task PATCH /transactions requests sent by Kotlin Reference Server were replaced with corresponding RPC calls

### Known limitations

[TODO or N/A]

### CODE REVIEW

File: kotlin-reference-server/src/main/resources/default-config.yaml [hunk 1]
```diff
@@ -14,3 +14,5 @@ sep24:
   interactiveJwtKey: "secret_sep24_interactive_url_jwt_secret"
   # Indicates, that custody integration is enabled
   custodyEnabled: false
+  # Indicates, that RPC actions should be used instead of PATCH /transactions endpoint
+  rpcActionsEnabled: false
```
Comment (rkharevych) [inline kotlin-reference-server/src/main/resources/default-config.yaml:18]: What default value should we use?

**Review by MazurakIhor — APPROVED** [review]

**Comment by rkharevych** [comment]
Hi, @Ifropc.
Do we need to update `anchor_reference server` because `AnchorEventProcessor` uses `PATCH /transactions` endpoint, but only to update status. How should we map these requests to actions, since actions require body or we can hardcode some stub values?

**Comment by Ifropc** [comment]
> Hi, @Ifropc. Do we need to update `anchor_reference server` because `AnchorEventProcessor` uses `PATCH /transactions` endpoint, but only to update status. How should we map these requests to actions, since actions require body or we can hardcode some stub values? This PR will be updated after we merge #1013

Java reference server is being deprecated soon, so it's fine to keep patch transactions calls there

File: kotlin-reference-server/src/main/kotlin/org/stellar/reference/sep24/DepositService.kt [hunk 2]
```diff
@@ -78,29 +76,84 @@ class DepositService(private val cfg: Config) {
     val fee = calculateFee(amount)
     val stellarAsset = "stellar:$asset"
 
-    sep24.patchTransaction(
-      PatchTransactionTransaction(
+    if (cfg.sep24.rpcActionsEnabled) {
+      sep24.rpcAction(
+        "request_offchain_funds",
+        buildJsonObject {
```
Comment (Ifropc) [inline kotlin-reference-server/src/main/kotlin/org/stellar/reference/sep24/DepositService.kt:82]: Can you create a `@Serializable` data class for that?

File: kotlin-reference-server/src/main/kotlin/org/stellar/reference/data/Data.kt [hunk 3]
```diff
@@ -38,6 +39,14 @@ data class PatchTransactionTransaction(
   @SerialName("memo_type") val memoType: String? = null,
 )
 
+@Serializable
+data class RpcRequest(
+  val id: String,
+  val jsonrpc: String,
+  val method: String,
+  val params: JsonObject
```
Comment (Ifropc) [inline kotlin-reference-server/src/main/kotlin/org/stellar/reference/data/Data.kt:47]: Can't it just be `Any`? Or (preferably) can you make it generic?

File: kotlin-reference-server/src/main/resources/default-config.yaml (hunk 1 above)
Comment (Ifropc) [reply kotlin-reference-server/src/main/resources/default-config.yaml:18]: Let's set it `true` by default

**Pull request merged**